registry/purview-registry/registry/purview_registry.py

```diff
@@ -0,0 +1,467 @@
+        derived_feature_entity = AtlasEntity(
+            name=definition.name,
+            qualified_name=qualified_name,
+            attributes={k:v for k,v in attrs.items() if k not in ['name','qualifiedName']},
+            typeName=str(EntityType.DerivedFeature),
+            guid=self.guid.get_guid())
+        self._upload_entity_batch(
+            [derived_feature_entity])
+        
+        # change from AtlasEntity to Entity
+        project_entity = self.get_entity(project_id)
+        derived_feature_entity = self.get_entity(derived_feature_entity.guid)
+
+        feature_project_contain_belong_pairs = self._generate_relation_pairs(
+            project_entity, derived_feature_entity, Label_Contains)
+
+        consume_produce_pairs = []
+        for input_feature in input_features:
+            consume_produce_pairs += self._generate_relation_pairs(
+                    derived_feature_entity, input_feature,Label_Consumes)
+
+        self._upload_entity_batch(
+            feature_project_contain_belong_pairs
+            + consume_produce_pairs)
+        
+        return derived_feature_entity.id
+    def _bfs(self, id: UUID, conn_type: RelationshipType) -> Tuple[list[Entity], list[Edge]]:
+        """
+        Breadth first traversal
+        Starts from `id`, follow edges with `conn_type` only.
+
+        WARN: There is no depth limit.
+        """
+        id_to_process = [id]
+        entity_ids = [id]
+        edges = []
+
+        while len(id_to_process)!=0:
+            outbound_edges = self._bfs_step(id_to_process,conn_type)
+            edges += outbound_edges
+            next_step_ids = list(set([x.to_id for x in outbound_edges]))
+            entity_ids.extend(next_step_ids)
+            entity_ids = list(set(entity_ids))
+            id_to_process = next_step_ids
+        
+        entities = self.get_entities(entity_ids,True)
+        return (entities,edges)
+
+        
+    
+    def _bfs_step(self, ids: list[UUID], conn_type: RelationshipType) -> list[Edge]:
+        """
+        One step of the BFS process
+        Returns all edges that connect to node ids the next step
+        """
+        return list(itertools.chain(*[self.get_neighbors(id,conn_type) for id in ids]))
+
+            
+
+    def _register_feathr_feature_types(self):
+        """
+        Register the feathr types if we haven't done so. Note that this only needs to be called once per provisioning
+        a system. Basically this function registers all the feature type definition in a Atlas compatible system.
+        """
+
+        # Each feature is registered under a certain Feathr project. The project should what we refer to, however for backward compatibility, the type name would be `feathr_workspace`
+        type_feathr_project = EntityTypeDef(
+            name=str(EntityType.Project),
+            attributeDefs=[
+                # "anchor_features" and "derived_features" are removed, since we are moving to use process entity
```

Comment:
Can we not remove those fields from the definition? We can just ignore those fields, as removing those attributes will need end users completely purge all the existing purview entities.

Reply:
Ok, will put empty properties instead